doc/source/tutorial-edit.rst

```diff
@@ -6,7 +6,116 @@ Editing rasters
+contain information about 'no data' values. If there is no existing 'no data' information ``rasterio`` will
+set 'no data' as ``nan``. In this image, ``nans`` do not exist, and we can see that because the 'dtype' is
+'uint16', whereas ``nans`` require data as floating point numbers.
+
+Let's save a temporary file below and specify the 'no data' value as 0. Then, when we open the temporary
+file the 'no data' attributes should be set as 0.
+
+.. ipython:: python
+
+    import tempfile
+    from pathlib import Path
+    import geowombat as gw
+    from geowombat.data import l8_224078_20200518
+
+    with gw.open(l8_224078_20200518) as src:
+        # Create a temporary directory
+        with tempfile.TemporaryDirectory() as tmp:
+            # Set the temporary file name
+            tmp_file = Path(tmp) / 'tmp_raster.tif'
+            # Save the data to file, setting 'no data' as 0
+            src.gw.save(tmp_file, nodata=0)
+            # Open the temporary file and view the data
+            with gw.open(tmp_file) as src_nodata:
+                print(src_nodata)
+                print(src_nodata.squeeze().values[0])
+
+.. note::
+
+    We are not modifying any data -- we are only updating the ``DataArray`` metadata. Thus, the printout of
+    the data above reflect changes in the ``DataArray`` 'no data' attributes but the printed array values
+    remained unchanged.
+
+But what if we want to modify the 'no data' value when opening the file (instead of re-saving)? We can
+pass ``nodata`` to the opener as shown below.
+
+.. ipython:: python
+
+    import geowombat as gw
+    from geowombat.data import l8_224078_20200518
+
+    with gw.open(l8_224078_20200518, nodata=0) as src:
+        print(src)
+        print(src.squeeze().values[0])
+
+We can also set 'no data' using the configuration manager like:
+
+.. ipython:: python
+
+    import geowombat as gw
+    from geowombat.data import l8_224078_20200518
+
+    with gw.config.update(nodata=0):
+        with gw.open(l8_224078_20200518) as src:
+            print(src)
+            print(src.squeeze().values[0])
+
+Masking 'no data' values
+------------------------
+
+As mentioned above, the array data are not automatically modified by the 'no data' value. If we want to
+mask our 'no data' values (i.e., exclude them from any calculations), we simply need to convert the
+array values to ``nans``. GeoWombat provides a method called :func:`DataArray.gw.mask_nodata` to do this
+that uses the metadata.
+
+.. ipython:: python
+
+    import geowombat as gw
+    from geowombat.data import l8_224078_20200518
+
+    with gw.open(l8_224078_20200518, nodata=0) as src:
+        # Print the first band with no masking
+        print('No masking:')
+        print(src.sel(band=1).values)
+        # Print the first band with masked 'no data'
+        print("\n'No data' values masked:")
+        print(src.gw.mask_nodata().sel(band=1).values)
```

Comment:
Ok this is very helpful! I have been wondering the same thing

Reply:
I am assuming this method works for all dtypes? uint8 etc?

Reply:
Yes, if the `nodata` value is `nan` (i.e., float), then the `DataArray` will be converted to a float before setting nans.

Reply:
For integers, the 'no data' value could be set as anything within the valid range. However, if the user set `nodata=32768` on an 'uint8' data, nothing would be set. Do we want to change this behavior @mmann1123?

Reply:
Since nans are the common value to use for masking arrays and the only way to insert nans is to ensure a data is of a floating type, I prefer to automatically convert the data dtype when `nodata=np.nan`. However, I would lean away from converting the data for other integer 'no data' values and would expect the user to manage the dtype.

Reply:
Yeah I don't love the way python handles missing values for int.  Maybe we can just print a warning if they set the nodata value to something outside the valid range?

Reply:
Check out the new comment on `geoxarray.mask_nodata()` and see if that makes sense.